### PR TITLE
adjust doc for new pod template support

### PR DESCRIPTION
**Pull request by gabemontero**
for https://trello.com/c/hS9lzo1v/1202-5-better-jenkins-slave-template-configuration-jenkinsintegration

@bparees @bmcelvee

### CODE REVIEW

**Comment by gabemontero** [comment]
[(5) Better jenkins slave template configuration [jenkins_integration]](https://trello.com/c/hS9lzo1v/1202-5-better-jenkins-slave-template-configuration-jenkinsintegration)

File: using_images/other_images/jenkins.adoc [hunk 1]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
 
 The Jenkins image also provides auto-discovery and auto-configuration
 of slave images for the
-Kubernetes plug-in. The Jenkins image searches for these in the existing image streams within the project that it is running in. The search specifically looks for image streams that have the label `role` set to `jenkins-slave`.
+Kubernetes plug-in. The Jenkins image, via the link:https://github.com/openshift/jenkins-sync-plugin[OpenShift Sync plug-in],
+on Jenkins start-up searches for these in the existing image streams within the project that it is running in, along with any additional
+projects listed in the plug-in's configuration. The search specifically looks for:
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: it's actually either it searches the project it's running in, OR it searches the projects listed in the configuration, i believe.  If you want to search your own project and another one, you need to list both.

File: using_images/other_images/jenkins.adoc [hunk 2]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
...
 
-When it finds an image stream with this label, it generates the corresponding Kubernetes plug-in
+* image streams that have the label `role` set to `jenkins-slave`.
+* image stream tags that have the annotation `role` set to `jenkins-slave`.
+* config maps that have the label `role` set to `jenkins-slave`.
+
+When it finds an image stream with the appropriate label, or image stream tag with the appropriate  annontation, it generates the corresponding Kubernetes plug-in
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: typo on annotation

File: using_images/other_images/jenkins.adoc [hunk 3]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
...
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: v1
+  data:
+    asdf: |-
+      <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+        <inheritFrom></inheritFrom>
+        <name>asdf</name>
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: better sample name please

File: using_images/other_images/jenkins.adoc [hunk 4]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
...
+        <instanceCap>2147483647</instanceCap>
+        <idleMinutes>0</idleMinutes>
+        <label>asdf</label>
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: same

File: using_images/other_images/jenkins.adoc [hunk 5]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
...
+
+The following is an example config map:
+
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: v1
+  data:
+    asdf: |-
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: better sample key

File: using_images/other_images/jenkins.adoc [hunk 6]
```diff
@@ -425,17 +425,109 @@ scripts for your image.
...
+            <resourceLimitMemory></resourceLimitMemory>
+            <envVars/>
+          </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
+        </containers>
+        <envVars/>
+        <annotations/>
+        <imagePullSecrets/>
+        <nodeProperties/>
+      </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+    fdsa: |-
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: same here and in name/label

**Review by bparees — CHANGES_REQUESTED** [review]

File: using_images/other_images/jenkins.adoc (hunk 1 above)
Comment (gabemontero) [inline using_images/other_images/jenkins.adoc]: yep misread the code earlier

**Comment by gabemontero** [comment]
updates pushed

**Review by bparees — APPROVED** [review]

**Comment by bparees** [comment]
lgtm

**Comment by bmcelvee** [comment]
LGTM

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../using_images/other_images/jenkins.adoc#using-images-other-images-jenkins[Jenkins]
|Added content to the xref:../using_images/other_images/jenkins.adoc#using-the-jenkins-kubernetes-plug-in-to-run-jobs[Using the Jenkins Kubernetes Plug-in to Run Jobs] section for new pod template support.
%

**Comment by dongboyan77** [comment]
@gabemontero hi, do you think we need to add some descriptions for annotation `"slave-label": "<label>"` in imagestream and imagestreamTag

**Comment by gabemontero** [comment]
hey @dongboyan77 - good catch!  Forgot about that nuance.  I'll submit a follow up PR.

**Pull request merged**